modules/ballerina-core/src/main/java/org/ballerinalang/services/dispatchers/DispatcherRegistry.java

```diff
@@ -44,6 +46,10 @@ public ServiceDispatcher getServiceDispatcher(String protocol) {
         return serviceDispatchers.get(protocol);
     }
 
+    public ServiceDispatcher getServiceDispatcherFromPkg(String protocolPkg) {
```

Comment:
There is already a method with that name, which uses protocol as a parameter(so same signature) and it is used in the dispatching logic where it finds the correct dispatcher using protocol(in here we cannot use protocol packagePath as it was set in transport level)